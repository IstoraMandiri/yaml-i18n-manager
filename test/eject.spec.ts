import rimraf from 'rimraf';
import fs from 'fs-extra';

import eject from '../src/eject';

import translatedYaml from './data/misc/translated.json';
import scan from './util/scan';

const outDir = `./test/data/out-temp`;
const mergeSrc = './test/data/merge';
const mergeTest = './test/data/merge-test';

function clean() {
  rimraf.sync(outDir);
}
describe('eject', () => {
  beforeAll(clean);
  afterEach(clean);
  it('ejects freshly correctly', async () => {
    await eject(translatedYaml, outDir);
    expect(await scan(outDir)).toMatchSnapshot();
  });
  describe('merging', () => {
    beforeEach(async () => {
      rimraf.sync(mergeTest);
      await fs.copy(mergeSrc, mergeTest);
    });
    afterEach(async () => {
      rimraf.sync(mergeTest);
    });
    it('merges correctly', async () => {
      await eject(translatedYaml, mergeTest);
      expect(await scan(mergeTest)).toMatchSnapshot();
    });
  });
});
