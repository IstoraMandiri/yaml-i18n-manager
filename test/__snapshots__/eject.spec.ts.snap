// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eject ejects freshly correctly 1`] = `
Array [
  Object {
    "content": "home:
  title: ホーム
about:
  title: 約
news:
  title: ニュース
",
    "fileName": "test/data/out-temp/menu.ja.yaml",
  },
  Object {
    "content": "intro: このサブフォルダへようこそ
description: これは翻訳されない説明です
items:
  test:
    title: テスト中
  hello:
    title: ハローワーク
",
    "fileName": "test/data/out-temp/sub/welcome.ja.yaml",
  },
  Object {
    "content": "title: 題名
description: これは説明です
new: 新しいキー
body: |
  ##これはいくつかの値下げです

  これはどのように？
",
    "fileName": "test/data/out-temp/ui.global.ja.yaml",
  },
]
`;

exports[`eject merging merges correctly 1`] = `
Array [
  Object {
    "content": "home:
  title: ホーム
about:
  title: 約
news:
  title: ニュース
special:
  title: Don't delete me
",
    "fileName": "test/data/merge-test/menu.ja.yaml",
  },
  Object {
    "content": "intro: このサブフォルダへようこそ
description: これは翻訳されない説明です
items:
  \\"0\\":
    key: hello
    title: ハローワーク
  test:
    title: テスト中
  hello:
    title: ハローワーク
",
    "fileName": "test/data/merge-test/sub/welcome.ja.yaml",
  },
  Object {
    "content": "title: 題名
description: これは説明です
new: 新しいキー
body: |
  ##これはいくつかの値下げです

  これはどのように？
",
    "fileName": "test/data/merge-test/ui.global.ja.yaml",
  },
]
`;
